packages/databyss-ui/components/Editor/state/reducer.js

```diff
@@ -64,6 +67,74 @@ const setActiveBlockType = (state, type, isNew) => {
   }
 }
 
+const insertBlock = (state, payload) => {
+  const {
+    activeBlockId,
+    activeBlockText,
+    previousBlockId,
+    previousBlockText,
+  } = payload.blockProperties
+
+  let _state = state
```

Comment:
This smells bad. It implies that `_state` and `state` point to different objects, when they don't. You should change it to `_state = cloneDeep(state)` because you are changing the `.page.blocks` leaf "in-place" on 79

Reply:
i'll do this one